src/libraries/context-recording-selection/RecordingSelectionContext.ts

```diff
@@ -325,8 +329,8 @@ const zoomTime = (state: RecordingSelection, action: ZoomRecordingSelectionActio
 }
 
 const setFocusTime = (state: RecordingSelection, action: SetFocusTimeRecordingSelectionAction): RecordingSelection => {
-    const {focusTimeSec, autoScrollVisibleTimeRange} = action
-    const newState = { ...state, focusTimeSeconds: focusTimeSec }
+    const {focusTimeSec, shiftKey, autoScrollVisibleTimeRange} = action
```

Comment:
Better not to mix the UI implementation detail (whether the shift key being held down) with the internal state logic (whether to set focus time or focus interval).

I think we'd be better served here by having a separate reducer action to set the interval (and maybe it takes both ends of the interval, or is overloaded so that it infers that one of them is the current focus time if it only gets one time) and let the front-end look at the shift key to choose which to call. (This would also make it easier to implement a drag-based version if we changed our minds or wanted to make it easier for the user to modify an existing interval.)